src/ray/core_worker/core_worker.cc

```diff
@@ -159,7 +159,15 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
   Status raylet_client_status;
   NodeID local_raylet_id;
   int assigned_port;
-  std::string serialized_job_config = options_.serialized_job_config;
+
+  if (options_.worker_type == WorkerType::DRIVER &&
+      !options_.serialized_job_config.empty()) {
+    // Driver populates job_config through worker startup options.
```

Comment:
IIUC driver is not started with worker startup options? 

Maybe it should be "driver populates the job config via initialization. Workers populates it when the first task is received"?